libs/wingc/src/jsify.rs

```diff
@@ -1534,3 +1562,58 @@ impl<'a> FieldReferenceVisitor<'a> {
 		}
 	}
 }
+
+/// Visitor that finds all the types defined in preflight that
+/// are referenced inside a scope (used on inflight methods)
+struct PreflightTypeRefVisitor<'a> {
+	/// Set of user types referenced inside the method
+	references: IndexMap<String, TypeRef>,
+
+	/// The root scope of the function we're analyzing
+	function_scope: &'a Scope,
+
+	/// The current env, used to lookup the type
+	env: SymbolEnvRef,
+}
+
+impl<'a> PreflightTypeRefVisitor<'a> {
+	pub fn new(function_scope: &'a Scope) -> Self {
+		Self {
+			references: IndexMap::new(),
+			function_scope,
+			env: function_scope.env.borrow().as_ref().unwrap().get_ref(),
+		}
+	}
+
+	pub fn find_preflight_type_refs(mut self) -> IndexMap<String, TypeRef> {
+		self.visit_scope(self.function_scope);
+		self.references
+	}
+}
+
+impl<'ast> Visit<'ast> for PreflightTypeRefVisitor<'ast> {
+	fn visit_scope(&mut self, node: &'ast Scope) {
+		self.env = node.env.borrow().as_ref().unwrap().get_ref();
+		visit::visit_scope(self, node);
```

Comment:
Should we "restore" `self.env` to the previous env after we're done visiting a child scope? Or add a comment maybe...

Reply:
good catch, fixed.